pkg/controller/common.go

```diff
@@ -1529,29 +1527,44 @@ func (c *SharedState) createReplicas(ctx context.Context, remainingReplicas int,
 		remainingReplicas--
 	}
 
+	for _, podObj := range podObjs {
+		azr.getSharedState().eventRecorder.Eventf(podObj.DeepCopyObject(), v1.EventTypeNormal, ReplicaAttachmentCreationSuccessEvent, "%d replica attachment(s) created for volume %s", requiredReplicas-remainingReplicas, volumeName)
```

Comment:
The replica `AzVolumeAttachment` instance has been created at this point, but the disk attach operation has not yet completed. We should generate a node event when the attachment has completed, e.g. "Replica mount of volume %s successfully attached to node %s" on success and "Replica mount of volume %s failed to attach on node %s with error %q" on error.

Reply:
Should we also generate a PV event for that or would that be unnecessary?

Reply:
I don't believe the PV would be the right target and it doesn't need to be done in both places. The pod is the one requesting the PV be mounted to the primary and replicas. I expected a customer would look at the pod events to determine whether the expected number of replicas had been created and attached.

Reply:
(Removed "node" from my original comment)

Reply:
Publishing the success & failure events from attach-detach controller now.